feat: render Solr highlights as bold in instrument list

PouyaMohseni opened the pull request:
- Added `apply_highlights` method to `InstrumentNameSet` to apply Solr highlight snippets to each instrument name.
- Replaced matching parts of instrument names with highlighted `<b>` text from Solr.
- Updated `SolrInstrument` to store highlights and pass them to `InstrumentNameSet`.
- Updated template to render instrument names with applied highlights.

resolves #343

<img width="947" height="748" alt="example" src="https://github.com/user-attachments/assets/6178f6f8-7688-413d-9077-628cdf29ae8d" />

yinanazhou commented on web-app/django/VIM/apps/instruments/views/instrument_list.py:
```diff
@@ -261,14 +281,36 @@ def _get_solr_page_results(
-        ]
+
+        # Extract highlight info
+        highlights = getattr(solr_response, "highlighting", {})
```
I personally prefer to rename `highlights` to `highlight_info` or `highlight_data`.

yinanazhou commented on web-app/django/VIM/apps/instruments/views/instrument_list.py:
```diff
@@ -261,14 +281,36 @@ def _get_solr_page_results(
+        instruments = []
+        for doc in solr_response.docs:
+            pk = doc.get("sid", "").replace("instrument-", "")
+            hl_snippets = highlights.get(f"instrument-{pk}", {}).get("text", [])
```
`hl_snippets` contains lots of empty strings. Filter them out first?

PouyaMohseni committed:
feat: render Solr highlights as bold in instrument list

- Added `apply_highlights` method to `InstrumentNameSet` to apply Solr highlight snippets to each instrument name.
- Replaced matching parts of instrument names with highlighted `<b>` text from Solr.
- Updated `SolrInstrument` to store highlights and pass them to `InstrumentNameSet`.

resolves #343

PouyaMohseni committed:
feat: render Solr highlights as bold in instrument list

- Added `apply_highlights` method to `InstrumentNameSet` to apply Solr highlight snippets to each instrument name.
- Replaced matching parts of instrument names with highlighted `<b>` text from Solr.
- Updated `SolrInstrument` to store highlights and pass them to `InstrumentNameSet`.

resolves #343

PouyaMohseni committed:
feat: update template to render instrument names with applied highlights

PouyaMohseni committed:
feat: update template to render instrument names with applied highlights
